clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2342,16 +2343,56 @@ void handleFailure(Throwable throwable) {
         return new DescribeConsumerGroupsResult(new HashMap<String, KafkaFuture<ConsumerGroupDescription>>(futures));
     }
 
+    private final static class ListConsumeGroupsResults {
+        private final List<Throwable> errors;
+        private final HashMap<String, ConsumerGroupListing> listings;
```

Comment:
This does not need to be a map, a list is good enough since we would not call `addList` with the same groupId, similar to `addError`.

Reply:
Isn't it possible that two nodes could both think that they own the same group?

Reply:
Probably a tough case to hit in practice, but I can't say it's impossible.